### PR TITLE
Add maintenance.md for supported Kubernetes upgrade

### PR DESCRIPTION
**Pull request by tapih**
We need some special knowledge for upgrading the supported Kubernetes versions.
This PR adds a document to decrease the knowledge gap between maintainers about how to upgrade the versions.
Signed-off-by: H.Muraoka <hiroshi-muraoka@cybozu.co.jp>

### CODE REVIEW

File: docs/maintenance.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+How to upgrade supported Kubernetes version
+===========================================
+
+TopoLVM depends on some Kubernetes repositories like `k8s.io/client-go` and should support 3 Kubernets versions at a time.
+Here is the guide for how to upgrade the supported versions.
+
+### Check release notes
```
Comment (ymmt2005) [inline docs/maintenance.md:7]:
```suggestion
## Check release notes
```

Please adjust other titles as well.

File: docs/maintenance.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+- `example/Makefile`: Makefile for running example
+- `deploy/manifests/overlays/daemonset-scheduler/kustomization.yaml`: Kustomization for overwriting the TopoLVM image version
+- `deploy/manifests/overlays/deployment-scheduler/kustomization.yaml`: Kustomization for overwriting the TopoLVM image version
+
+`git grep 1.18`, `git grep image:`, and `git grep VERSION` might help us avoid overlooking necessary changes.
+Please update the versions in the code and docs with great care.
+
+### Update CSI spec
+
+In case we update the version of the CSI spec written in `Makefile`, we should regenerate the gRPC schema with the protobuf file corresponding to the CSI spec version.
```
Comment (ymmt2005) [inline docs/maintenance.md:57]: To update **CSI** version, we need to update `CSI_VERSION` in Makefile, and run the following commands:

```console
$ make csi.proto
$ make generate 
```

**Review by ymmt2005 — CHANGES_REQUESTED** [review]

**Review by ymmt2005 — APPROVED** [review]
LGTM

**Review by toshipp — DISMISSED** [review]
LGTM.
I'm waiting for @llamerada-jp 's review before merge.

**Review by llamerada-jp — APPROVED** [review]
LGTM

**Review by toshipp — APPROVED** [review]

**Pull request merged**